employee-rostering-frontend/src/ui/components/DataTable.tsx

```diff
@@ -0,0 +1,178 @@
+        <CloseIcon />
+      </Button>
+    </span>;
+  }
+
+  getEditButtons(dataStore: C, data: T, editableComponents: EditableComponent[]): JSX.Element {
+    return <EditableComponent ref={(c) => {editableComponents[editableComponents.length-1] = c as EditableComponent;}}
+      viewer={<span>
+        <Button aria-label="Edit"
+          variant={ButtonVariant.link}
+          onClick={() => {
+            editableComponents.forEach(c => c.startEditing());
+          }}>
+          <EditIcon />
+        </Button>
+        <Button aria-label="Delete"
+          variant={ButtonVariant.link}
+          onClick={() => {
+            this.removeData(data);
+          }}>
+          <TrashIcon />
+        </Button>
+      </span>}
+      editor={<span>
+        <Button aria-label="Save"
+          variant={ButtonVariant.link}
+          onClick={() => {
+            if (this.isValid(dataStore)) {
+              this.updateData(this.extractDataFromRow(data,dataStore));
+              editableComponents.forEach(c => c.stopEditing());
+            }
+          }}>
+          <SaveIcon />
+        </Button>
+        <Button aria-label="Cancel"
+          variant={ButtonVariant.link}
+          onClick={() => {
+            editableComponents.forEach(c => c.stopEditing());
+          }}>
+          <CloseIcon />
+        </Button>
+      </span>}/>;
+  }
+
+  convertDataToTableRow(data: T): IRow {
+    const dataStore: any = {};
+    const viewers = this.displayDataRow(data);
+    const editors = this.editDataRow(dataStore, data);
+    const length = viewers.length
+    const editableComponents: EditableComponent[] = new Array(length + 1);
+    const cellContents = viewers.map((viewer, index) => {return { title:
+      <EditableComponent viewer={viewer}
+        editor={editors[index]}
+        ref={(c) => editableComponents[index] = c as EditableComponent}
+      />} }).concat([{title: this.getEditButtons(dataStore, data, editableComponents)}]);
+    return {
+      cells: cellContents
+    };
+  }
+
+  // Use for SNAPSHOT testing
+  renderViewer(data: T): JSX.Element {
+    return <tr>{this.displayDataRow(data).map((c,index) => <td key={index}>{c}</td>)}</tr>;
+  }
+
+  // Use for SNAPSHOT testing
+  renderEditor(data: T): JSX.Element {
+    const dataStore: any = {};
+    return <tr>{this.editDataRow(dataStore, data).map((c,index) => <td key={index}>{c}</td>)}</tr>;
+  }
+
+  render() {
+    const additionalRows: IRow[] = (this.state.newRow !== null)? [this.state.newRow] : [];
+    const rows = additionalRows.concat(this.props.tableData
+      .filter(this.state.currentFilter).map(this.convertDataToTableRow));
+    const columns = this.props.columnTitles.map(t => { 
+      return { title: t, cellTransforms: [headerCol], props: {} };
+    }).concat([{title: '', cellTransforms: [headerCol], props: {}}]);
+    return (
+      <div>
+        <Button disabled={this.state.newRow !== null} onClick={this.createNewRow}>Add</Button>
```

Comment:
Should be `isDisabled`. `disabled` is a prop representing an HTML attribute but we want to use the PaterrnFly `Button`'s prop, which is called `isDisabled`.